Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -122,10 +110,11 @@ public struct NIOIsolatedEventLoop {
         line: UInt = #line,
         _ task: @escaping () throws -> EventLoopFuture<T>
     ) -> Scheduled<T> {
-        let unsafeTransfer = UnsafeTransfer(task)
-        return self._wrapped.flatScheduleTask(deadline: deadline, file: file, line: line) {
-            try unsafeTransfer.wrappedValue()
-        }
+        let promise: EventLoopPromise<T> = self._wrapped.makePromise(file: file, line: line)
+        let scheduled = self.scheduleTask(deadline: deadline, task)
+
+        scheduled.futureResult.flatMap { $0 }.cascade(to: promise)
```

Comment:
Can we save an alloc here if we use `scheduled.futureResult.whenComplete { ... }`?

Reply:
I don't think so: the `flatMap` unwraps an extra level of futureness.

Reply:
I mean like this (I appreciate it's more code than the `flatMap` + `cascade`):

```swift
scheduled.futureResult.whenComplete { result in
    switch result {
    case .success(let futureResult):
        promise.completeWith(futureResult)
    case .failure(let error):
        promise.fail(error)
    }
}
```

Reply:
Yes, you're probably right. However, all the `flatScheduleTask` implementations in swift-nio follow this spelling. I've filed #3071 to track making the improvement generically, and so for now would rather keep this in line with the others.

Reply:
Thanks for filing that, makes sense to track them together.